src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesResource.kt

```diff
@@ -1,18 +1,37 @@
 package uk.gov.justice.digital.hmpps.prisonertonomisupdate.incentives
 
 import com.microsoft.applicationinsights.TelemetryClient
+import kotlinx.coroutines.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 import org.springframework.http.HttpStatus
 import org.springframework.web.bind.annotation.PutMapping
 import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestController
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.config.trackEvent
 
 @RestController
-class IncentivesResource(private val telemetryClient: TelemetryClient) {
+class IncentivesResource(private val telemetryClient: TelemetryClient, private val incentivesReconciliationService: IncentivesReconciliationService) {
+
+  private companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @OptIn(DelicateCoroutinesApi::class)
   @PutMapping("/incentives/reports/reconciliation")
   @ResponseStatus(HttpStatus.ACCEPTED)
   fun generateIncentiveReconciliationReport() {
     telemetryClient.trackEvent("incentives-reports-reconciliation-requested")
-    telemetryClient.trackEvent("incentives-reports-reconciliation-report", mapOf("status" to "success", "issues-count" to "0"))
+    log.info("Incentives reconciliation report requested")
+    GlobalScope.launch {
```

Comment:
run is coroutine - fire and forget

Reply:
Have you read this? https://elizarov.medium.com/the-reason-to-avoid-globalscope-835337445abc